src/api/app/controllers/source_package_command_controller.rb

```diff
@@ -14,6 +14,7 @@ class SourcePackageCommandController < SourceController
   # we use an array for the "file" parameter
   skip_before_action :validate_params, only: %i[diff linkdiff servicediff]
 
+  before_action :set_project, only: :undelete
   before_action :set_target_project_name
```

Comment:
```suggestion
  before_action :set_project, only: :undelete
  before_action :set_target_project_name, except:  :undelete
```

Reply:
Not now, still need this for `before_action :authorize`....